azure/services/groups/groups.go

```diff
@@ -35,81 +33,42 @@ const ServiceName = "group"
 // Service provides operations on Azure resources.
 type Service struct {
 	Scope GroupScope
-	aso.Reconciler[ASOType]
+	*aso.Service[ASOType, GroupScope]
 }
 
 // GroupScope defines the scope interface for a group service.
 type GroupScope interface {
 	azure.AsyncStatusUpdater
+	aso.Scope
 	GroupSpec() azure.ASOResourceSpecGetter[*asoresourcesv1.ResourceGroup]
-	GetClient() client.Client
-	ClusterName() string
 }
 
 // New creates a new service.
 func New(scope GroupScope) *Service {
+	svc := aso.NewService[ASOType](ServiceName, scope)
+	svc.Specs = []azure.ASOResourceSpecGetter[ASOType]{scope.GroupSpec()}
+	svc.PostReconcileHook = postReconcileHook
+	svc.PostDeleteHook = postDeleteHook
```

Comment:
Initializing a service is kind of clunky at the moment combining a constructor with setting individual fields.